pkg/utils/environment_variable.go

```diff
@@ -0,0 +1,58 @@
+package utils
+
+import (
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/log"
+)
+
+var defaultEnvVarValueOrDefault = envVarValueOrDefault
+
+type EnvVarValueOrDefaultFunc func(envVarName string, defaultValue string) string
+
+func IntFromEnvOrDefault(
+	envVarName string,
+	defaultValue string,
+	envVarValueOrDefaultFunc EnvVarValueOrDefaultFunc,
+) (int, error) {
+	if envVarValueOrDefaultFunc == nil {
```

Comment:
when would this not be `nil`? an example would help me understand what we are trying to accomplish here (:

Reply:
IMO this breaking up of parameters style is really hard for my 👀  to follow

Reply:
look at the UT :)

Reply:
this is the way we can pass mocks to the functions in go

Reply:
> IMO this breaking up of parameters style is really hard for my 👀 to follow

i understand. i had a conversation about this with @sgnn7 yesterday. the alternative is a line that is too long which is an issue as well.